### ISSUE TITLE
Bugs with ImageDataGenerator

### ISSUE THREAD

**Issue by jamesben6688** [fchollet/keras#8384]
I am using model.predict_generator and ImageDataGenerator to predict my image label, my code is as follows:
    `valid_gen= ImageDataGenerator().flow_from_directory("../input/valid/valid", target_size=(input_size, input_size),shuffle=False, batch_size=1)`
   `pred = model.predict_generator(generator=valid_gen, steps = 1, verbose=1)` 

actually I put only one image into folder "../input/valid/valid", then I check the predict result, I found it is totally wrong. I test for many images, so I think this is a bug. Anyone got some idea about this?

**Comment by jamesben6688** [comment]
I use model.fit_generator(train_gen.flow_from_directory(...)...) to train my model, then I try to use model.predict() to predict one single image, the result is totally different from model.predict_generator, and both labels are wrong.
